### ISSUE TITLE
`generate.py` and `generate/full.py` could be merged

### ISSUE THREAD

**Issue by carmocca** [Lightning-AI/lit-llama#313]
Proposed in https://github.com/Lightning-AI/lit-llama/pull/255

The only difference in logic is the instruction tuning. We could add a flag for it as in https://github.com/Lightning-AI/lit-llama/pull/278

**Comment by pavanpvl** [comment]
Hi @carmocca 
If possible I would like to work on this issue after going through the files and the discussions mentioned above what I can understand is as follows 

1. Remove the generate.py file and add the required changes to the generate/full.py file
2. Add instruction tuning flag and use generate_prompt function if the flag is true 
3. I can see that generate function in generate.py is referenced in multiple scripts(finetune, evaluate & generate) it can be moved to lit_llama/utils

Let me know if these changes sound good :D

**Comment by rasbt** [comment]
That's a good point. On that note, I wonder if we even need a `generate.py` script in the top level. Maybe we should just have all scripts under `generate/` similar to the `finetune/` and `evaluate/` scripts?

**Comment by pavanpvl** [comment]
@rasbt Yeah this sounds good as well, I was thinking of having it under utils but it might just be a matter of preference I guess 😅

**Comment by ruoyu61** [comment]
@carmocca I made a [PR](https://github.com/Lightning-AI/lit-llama/pull/432) for this issue. 

Could you help to review the PR?
